src/Package/Impl/DataInspect/Viewers/GridViewerBase.cs

```diff
@@ -32,7 +33,12 @@ internal abstract class GridViewerBase : ViewerBase, IObjectDetailsViewer {
             if (evaluation != null) {
                 await VsAppShell.Current.SwitchToMainThreadAsync();
 
-                var id = _toolWindowIdBase + evaluation.GetHashCode() % (Int32.MaxValue - _toolWindowIdBase);
+                var id = Math.Abs(_toolWindowIdBase + expression.GetHashCode() % (Int32.MaxValue - _toolWindowIdBase));
+
+                var existingPane = ToolWindowUtilities.FindWindowPane<VariableGridWindowPane>(id);
+                var frame = existingPane?.Frame as IVsWindowFrame;
+                frame?.CloseFrame(0);
```

Comment:
If it doesn't refresh, I'd just treat it as a separate bug to fix in another PR.